src/Http/RequestTransformer.php

```diff
@@ -48,6 +48,12 @@ public static function toCake(PsrRequest $request)
             $post = Hash::merge($post, $files);
         }
 
+        $input = null;
+        $stream = $request->getBody();
+        if ($stream->getSize()) {
```

Comment:
After knowing a bit of how a stream is implemented, I'd suggest just getting the content and then casting the `$input` to null if it is an empty string